components/apimgt/org.wso2.carbon.apimgt.rest.api.endpoint.registry/src/main/java/org/wso2/carbon/apimgt/rest/api/endpoint/registry/impl/RegistriesApiServiceImpl.java

```diff
@@ -161,9 +176,35 @@ public Response createRegistryEntry(String registryId, RegistryEntryDTO registry
                         " is not found", log);
             }
             ResourceFile definitionFile;
+            if ((definitionFileInputStream != null || registryEntry.getDefinitionUrl() != null) &&
+                    registryEntry.getDefinitionType() == null) {
+                RestApiUtil.handleBadRequest("Missing definitionType of the new registry " +
+                        "entry of the endpoint registry with id: " + registryId, log);
+            }
             if (definitionFileInputStream == null || definitionFileDetail == null) {
+                // Retrieve the endpoint definition from URL
+                try {
+                    URL definitionURL = new URL(registryEntry.getDefinitionUrl());
+                    if (!isValidEndpointDefinition(definitionURL, null,
+                            registryEntry.getDefinitionType().toString())) {
+                        RestApiUtil.handleBadRequest("Error while validating the endpoint URL definition of " +
```

Comment:
what about "Error while validating the endpoint URL definition of the new registry entry with registry id"

Reply:
updated